atomic_reactor/koji_util.py

```diff
@@ -40,6 +43,35 @@ def callback(self, offset, totalsize, size, t1, t2):  # pylint: disable=W0613
                               percent_done, size / t1 / 1024 / 1024)
 
 
+class KojiSessionWrapper(object):
+    """
+    Wrap all calls to koji.ClientSession methods in a catch/exception block, so that
+    improperly handled ConnectionErrors from koji.ClientSession will get retried silently.
+
+
+    """
+    def __init__(self, session):
+        self._wrapped_session = session
+
+    def __getattr__(self, name):
+        session_attr = getattr(self._wrapped_session, name)
+        if callable(session_attr):
+            def call_with_catch(*a, **kw):
+                retry_delay = HTTP_BACKOFF_FACTOR
+                last_exc = None
+                for retry in range(HTTP_MAX_RETRIES):
+                    try:
+                        return session_attr(*a, **kw)
+                    except ConnectionError as exc:
+                        time.sleep(retry_delay * (2 ** retry))
```

Comment:
This will go:
* call -> exception
* time.sleep() ...
* call -> exception
* time.sleep() ......
* call -> exception
* time.sleep() ..................
* propagate exception

In other words, we are sleeping *after* each attempt, even the last. Don't we want to sleep *before* each attempt other than the first?